docs/setup.md

```diff
@@ -2,7 +2,146 @@
 title: 'Setup'
 ---
 
-# GitHub Actions [linux/mac/windows] {: #github-actions}
+# Build your wheel locally
+
+Before starting to configure cibuildwheel, it's useful to try to build a wheel
+on your local machine. It's much easier to debug problems on your machine than
+inside a CI system!
+
+Start with a clean checkout of your project, inside a new clean virtual
+environment. Make sure your `pip` is up to date, and then invoke:
+
+```shell
+pip wheel -w wheelhouse .
+```
+
+If your build completes without a problem, you're in good shape! You can move on to
+the next step. Otherwise, you might have one of the following issues.
+
+### Missing build dependencies
+
+If your build needs Python dependencies, you can resolve this by adding
+package names to the
+[`build-system.requires`](https://www.python.org/dev/peps/pep-0518/#build-system-table)
+section of your pyproject.toml. For example, if your project requires Cython
+to build, your pyproject.toml might include a section like this:
+
+```toml
+[build-system]
+requires = [
+    "setuptools>=42",
+    "wheel",
+    "Cython",
+]
+
+build-backend = "setuptools.build_meta"
+```
+
+If you have other dependencies, you should make a note of these, you'll be
+able to install them using the cibuildwheel options [`CIBW_BEFORE_BUILD`](options.md#before-build) or
+[`CIBW_BEFORE_ALL`](options.md#before-all).
+
+### Actions you need to perform before building
+
+You might need to run some other commands before building, like running a
+script that performs codegen or downloading some data that's not stored in
+your source tree. There are a couple ways to deal with this:
+
+-   Incorporate this into your build process, by adding it to your package's
```

Comment:
This is setuptools specific, and does not apply to enscons and only partially to scikit-build. I'd probably add a little "if using setuptools" or similar.

Reply:
👍